.github/workflows/distro.yml

```diff
@@ -0,0 +1,272 @@
+        type: boolean
+        required: false
+        default: true
+      AIE_COMMIT:
+        description: 'which aie commit to build'
+        type: string
+        required: false
+        default: ''
+
+  pull_request:
+  release:
+    types:
+      - published
+
+jobs:
+
+  set_datetime:
+
+    name: Set canonical datetime
+
+    runs-on: ubuntu-latest
+
+    outputs:
+      DATETIME: ${{ steps.set_datetime.outputs.DATETIME }}
+
+    steps:
+      - name: set datetime
+        id: set_datetime
+        shell: bash
+        run: |
+          DATETIME=$(date +"%Y%m%d%H")
+          echo "DATETIME=${DATETIME}" | tee -a $GITHUB_OUTPUT
+
+  build_wheels:
+
+    needs: [set_datetime]
+
+    name: ${{ matrix.OS }} ${{ matrix.ARCH }}
+
+    continue-on-error: true
+
+    runs-on: ${{ matrix.OS }}
+
+    outputs:
+      MLIR_AIE_WHEEL_VERSION: ${{ steps.get_wheel_version.outputs.MLIR_AIE_WHEEL_VERSION }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - OS: ubuntu-20.04
+            ARCH: x86_64
+
+          - OS: ubuntu-20.04
+            ARCH: aarch64
+
+          - OS: macos-11
+            ARCH: x86_64
+
+          - OS: macos-11
+            ARCH: arm64
+
+          - OS: windows-2019
+            ARCH: AMD64
+
+    steps:
+    - uses: makslevental/mlir-wheels/.github/actions/setup_base@main
+      id: setup_base
+      with:
+        # optional
+        DEBUG_ENABLED: ${{ inputs.DEBUG_ENABLED }}
+        DEBUG_OS: ${{ inputs.DEBUG_OS }}
+        DEBUG_ARCH: ${{ inputs.DEBUG_ARCH }}
+        DEBUG_DETACHED: ${{ inputs.DEBUG_DETACHED }}
+        # required
+        MATRIX_OS: ${{ matrix.OS }}
+        MATRIX_ARCH: ${{ matrix.ARCH }}
+
+    - uses: makslevental/mlir-wheels/.github/actions/setup_ccache@main
+      id: setup_ccache
+      with:
+        MATRIX_OS: ${{ matrix.OS }}
+        MATRIX_ARCH: ${{ matrix.ARCH }}
+
+    - name: Checkout
+      working-directory: ${{ steps.setup_base.outputs.WORKSPACE_ROOT }}
+      shell: bash
+      run: |
+        
+        git clone --recursive https://github.com/Xilinx/mlir-aie.git
+        if [ x"${{ matrix.AIE_COMMIT }}" != x"" ]; then
+          pushd mlir-aie && git reset --hard ${{ matrix.AIE_COMMIT }} && popd
+        fi
+
+# https://stackoverflow.com/a/73467112
+    - name: Split os
+      shell: bash
+      env:
+        OS: ${{ matrix.OS }}
+      id: split
+      run: echo "platform_version=${OS##*-}" | tee -a $GITHUB_OUTPUT
+
+    - name: Install boost
+      uses: MarkusJx/install-boost@v2.4.4
+      id: install-boost
+      with:
+        boost_version: 1.72.0
+        platform_version: ${{ steps.split.outputs.platform_version }}
+        arch: ${{ matrix.ARCH == 'x86_64' && 'x86' || 'aarch64' }}
+
+    - name: cibuildwheel
+      if: ${{ matrix.OS != 'ubuntu-20.04' || matrix.ARCH != 'aarch64' }}
+      working-directory: ${{ steps.setup_base.outputs.WORKSPACE_ROOT }}
+      shell: bash
```

Comment:
Move to the top level?
https://github.com/Xilinx/mlir-aie/blob/main/.github/workflows/buildAndTest.yml#L16

Reply:
this can't be at top level because of 

```
    - name: rename
      if: ${{ matrix.OS == 'windows-2019' }}
      working-directory: ${{ steps.setup_base.outputs.WORKSPACE_ROOT }}
      run: |
        ls wheelhouse/mlir_aie-*whl | Rename-Item -NewName {$_ -replace 'cp311-cp311', 'py3-none' }
```

Reply:
Or the opposite with `shell: ...` for the Windows case?